control-base/src/main/java/fi/nls/oskari/control/layer/SaveLayerHandler.java

```diff
@@ -270,13 +323,21 @@ private boolean handleRequestToMapLayer(final ActionParameters params, OskariLay
             }
         }
 
-        // FIXME: Update JSON key when frontend implementation was updated
-        MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(params.getHttpParam("inspireTheme", -1));
-        ml.addGroup(maplayerGroup);
-
-        ml.setVersion(params.getHttpParam("version", ""));
+        String groupId = params.getHttpParam(PARAM_MAPLAYER_GROUPS, "-1");
+        ml.emptyMaplayerGroups();
+        if(groupId.contains(",")) {
+            String[] groupIds = groupId.split(",");
+            for (String groupid: groupIds) {
+                MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(ConversionHelper.getInt(groupid, -1));
+                ml.addGroup(maplayerGroup);
+            }
+        } else {
+            MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(ConversionHelper.getInt(groupId, -1));
+            ml.addGroup(maplayerGroup);
+        }
```

Comment:
You can remove the if-statement and just use split for any input since it returns an array of one element if comma isn't present. So instead of 10 lines you can just do:

    for (String id: groupId.split(",")) {
        MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(ConversionHelper.getInt(id, -1));
        ml.addGroup(maplayerGroup);
    }